src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -640,7 +642,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			let standardKernels = find(this._standardKernels, kernel => this._defaultKernel && kernel.displayName === this._defaultKernel.display_name);
 			let connectionProviderIds = standardKernels ? standardKernels.connectionProviderIds : undefined;
 			let providerFeatures = this._capabilitiesService.getCapabilities(profile.providerName);
-			if (connectionProviderIds.length > 0) {
+			if (connectionProviderIds.length > 0 && this._currentKernelAlias) {
```

Comment:
I don't quite understand this logic. Are we setting `this._currentKernelAlias` in another place as well?